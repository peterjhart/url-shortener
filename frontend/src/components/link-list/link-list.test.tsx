import { describe, expect, it } from 'vitest'
import { render } from '@testing-library/react'
import LinkList from './link-list'

describe('LinkList', () => {
  describe('Snapshots', () => {
    it('matches snapshot', () => {
      const links = [
        { id: '## ONE ##', alias: '## ALIAS ONE ##', url: '## URL ONE ##' },
        { id: '## TWO ##', alias: '## ALIAS TWO ##', url: '## URL TWO ##' },
        {
          id: '## THREE ##',
          alias: '## ALIAS THREE ##',
          url: '## URL THREE ##',
        },
      ]
      const { asFragment } = render(<LinkList links={links} />)

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
