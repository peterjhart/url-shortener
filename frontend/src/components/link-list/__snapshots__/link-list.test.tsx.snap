// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LinkList > Snapshots > matches snapshot 1`] = `
<DocumentFragment>
  <table
    class="table-auto w-full bg-gray-100"
  >
    <thead
      class="bg-gray-300"
    >
      <tr>
        <th
          class="bg-gray-300 rounded-tl-md text-left"
        >
          Alias
        </th>
        <th
          class="bg-gray-300 rounded-tr-md text-left"
        >
          Link
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="border border-t-0"
      >
        <td>
          ## ALIAS ONE ##
        </td>
        <td>
          ## URL ONE ##
        </td>
      </tr>
      <tr
        class="border border-t-0"
      >
        <td>
          ## ALIAS TWO ##
        </td>
        <td>
          ## URL TWO ##
        </td>
      </tr>
      <tr
        class="border border-t-0"
      >
        <td>
          ## ALIAS THREE ##
        </td>
        <td>
          ## URL THREE ##
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
